src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/CopiedCorHeaderNode.cs

```diff
@@ -62,75 +65,116 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             builder.RequireInitialPointerAlignment();
             builder.AddSymbol(this);
 
-            BlobReader reader = _module.PEReader.GetEntireImage().GetReader();
-            reader.Offset = _module.PEReader.PEHeaders.CorHeaderStartOffset;
-
-            // Header Size
-            int headerSize = reader.ReadInt32();
-            builder.EmitInt(headerSize);
-
-            // Runtime major, minor version
-            builder.EmitUShort(reader.ReadUInt16());
-            builder.EmitUShort(reader.ReadUInt16());
+            if (_module != null)
```

Comment:
Since we're splitting the function of this class in two based on the value of `_module`, we could have a `CompositeCorHeaderNode` and pick the right kind in `NodeFactory.CopiedCorHeader`